honeybee/radiance/datatype.py

```diff
@@ -10,61 +12,251 @@ class RadDescrDefault(object):
     method is not likely to be fancy most descriptors can inherit this class.
     """
 
+    def __init__(self, name, descriptiveName=None, expectedInputs=None,
+                 validRange=None):
+        self._name = "_" + name
+        self._descriptiveName = descriptiveName
+        self._expectedInputs = expectedInputs
+
+        if validRange:
+            assert isinstance(validRange, (tuple, list)) and len(
+                validRange) == 2, \
+                "The input for validRange should be a tuple/list containing" \
+                " expected minimum and maximum values"
+            validRange = sorted(validRange)
+
+        self._validRange = validRange
+
+        nameString = lambda: "%s (%s)" % (
+            name, descriptiveName) if descriptiveName else name
+
+        self._nameString = nameString()
+
     def __get__(self, instance, owner):
         value = getattr(instance, self._name)
         if value is None:
             raise Exception(
-                "The value for %s hasn't been specified" % self._name[1:])
+                "The value for %s hasn't been specified" % self._nameString)
         else:
             return getattr(instance, self._name)
 
+    def __set__(self, instance, value):
+        if self._expectedInputs and value is not None:
+
+            inputs = list(self._expectedInputs)
+            if value not in inputs:
+                raise ValueError("The value for %s should be one of the"
+                                 " following: %s. The provided value was %s"
+                                 % (self._nameString,
+                                    ",".join(map(str, inputs)), value))
+            setattr(instance, self._name, value)
 
-class RadDescrNumber(RadDescrDefault):
+class RadDescrBoolFlags(RadDescrDefault):
```

Comment:
@sariths I think the names of all the classes can be more human readable. Except for the base class we are not necessarily need to have Descr in the name. That's a technical aspect that can be hidden unless someone checks the source code. I suggest to change the names to RadinceBoolFlag, RadinceNumber, etc.